code/game/objects/items/toys.dm

```diff
@@ -1192,6 +1192,23 @@
 	var/cooldown = 0
 	resistance_flags = FLAMMABLE
 
+	COOLDOWN_DECLARE(catnip_delay)
+	///The catnip component for attracting cats
+	var/datum/component/catnip/cat_attractor // cattractor more like
+	///How long squeezing the cat toy summons the cats, and how long the delay to resqueeze it is afterwards
+	var/catnip_duration = 45 SECONDS
+
+/obj/item/toy/cattoy/attack_self(mob/user)
+	. = ..()
+	if(cat_attractor)
+		to_chat(user, "<span class='warning'>[src] is already emitting a catnip scent!</span>")
+		return
+	if(!COOLDOWN_FINISHED(src, catnip_delay))
+		to_chat(user, "<span class='warning'>[src] is still recharging its catnip banks!</span>")
+		return
+	to_chat(user, "<span class='notice'>You squeeze [src], giving off a burst of catnip!</span>")
+	AddComponent(/datum/component/catnip, scan_interval = 2.5 SECONDS, duration = catnip_duration)
```

Comment:
I feel like
A) this should be a signal
B) the catnip component should handle the cooldown and the attraction.
C) the catnip component should only be setup in the initalization of the item.